src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -51,15 +51,14 @@
 public interface RealmIdentity {
 
     /**
-     * Determine whether a given credential is definitely supported, possibly supported, or definitely not supported for this
-     * identity.  The credential type is defined by its {@code Class} and an optional {@code algorithmName}.  If the
-     * algorithm name is not given, then the query is performed for any algorithm of the given type.
+     * Determine whether a given credential is definitely obtainable, possibly obtainable, or definitely not obtainable for this
+     * identity.
      *
      * @param credentialName the name of the credential
      * @return the level of support for this credential type
      * @throws RealmUnavailableException if the realm is not able to handle requests for any reason
      */
-    CredentialSupport getCredentialSupport(String credentialName) throws RealmUnavailableException;
+    SupportLevel getCredentialSupport(String credentialName) throws RealmUnavailableException;
```

Comment:
I think (linguistically speaking) this method name now implies that if UNSUPPORTED is returned, the credential is unsupported across the board.  It might be better to rename this to getCredentialAcquisitionSupport() or getCredentialAcquireSupport(), and rename the other one to getEvidenceVerificationSupport() or getEvidenceVerifySupport().  Both grammatical forms are in wide use; either one would be fine with me (though all other things being equal, I lean towards shorter/more concise).